decoder.go

```diff
@@ -23,158 +20,99 @@ type (
 		Decode() (Event, error)
 	}
 	decoder struct {
-		r *bufio.Reader
-
-		// Buffers
-		eventID    *bytes.Buffer
-		eventType  *bytes.Buffer
-		dataBuffer *bytes.Buffer
-		field      *bytes.Buffer
-		value      *bytes.Buffer
+		scanner *bufio.Scanner
+		data    *bytes.Buffer
 	}
 )
 
-// NewDecoder creates a SSE decoder with the default buffer size.
+// NewDecoder builds an SSE decoder with a growing buffer.
+// Lines are limited to bufio.MaxScanTokenSize - 1.
 func NewDecoder(in io.Reader) Decoder {
-	return NewDecoderSize(in, defaultBufferSize)
+	return NewDecoderSize(in, 0)
 }
 
-// NewDecoderSize creates a SSE decoder with the specified buffer size.
-func NewDecoderSize(r io.Reader, bufferSize int) Decoder {
-	d := new(decoder)
-	d.initialise(r, bufferSize)
+// NewDecoderSize builds an SSE decoder with the specified buffer size (not growing).
+func NewDecoderSize(in io.Reader, bufferSize int) Decoder {
+	d := &decoder{bufio.NewScanner(in), new(bytes.Buffer)}
+	if bufferSize > 0 {
+		d.scanner.Buffer(make([]byte, bufferSize), bufferSize)
```

Comment:
If we have to get rid of this call to be backwards compatible I'd do it. The Scanner has a default buffer size of 4096 bytes. It's a pity it cannot be configured on older versions.